packages/cdktf-cli/lib/cdktf-project.ts

```diff
@@ -527,24 +534,49 @@ export class CdktfProject {
       if (runningStacks.length >= maxParallelRuns) {
         await Promise.race(runningStacks.map((s) => s.currentWorkPromise));
       } else {
-        const nextRunningExecutor = await next();
-        if (!nextRunningExecutor) {
-          // In this case we have no pending stacks, but we also can not find a new executor
-          break;
+        try {
+          const nextRunningExecutor = await next();
+          if (!nextRunningExecutor) {
+            // In this case we have no pending stacks, but we also can not find a new executor
+            break;
+          }
+          method === "deploy"
+            ? nextRunningExecutor.deploy(refreshOnly)
+            : nextRunningExecutor.destroy();
+        } catch (e) {
+          // await next() threw an error because a stack failed to apply/destroy
+          // wait for all other currently running stacks to complete before propagating that error
+          logger.debug(
+            "Encountered an error while awaiting stack to finish",
+            e
+          );
+          logger.debug("Waiting for still running stacks to finish");
```

Comment:
Nit: Maybe we can add the currently open stacks here for a better debugging log line

Reply:
done